content/docs/rfcs/12/README.md

```diff
@@ -61,77 +63,148 @@ The following are not considered as part of the adversarial model:
 ## Protobuf
 
 ```protobuf
-message FilterRequest {
-  bool subscribe = 1;
-  string topic = 2;
-  repeated ContentFilter contentFilters = 3;
-
-  message ContentFilter {
-    string contentTopic = 1;
+syntax = "proto3";
+
+// 12/WAKU2-FILTER rfc: https://rfc.vac.dev/spec/12/
+package waku.filter.v2;
+
+// Protocol identifier: /vac/waku/filter-subscribe/2.0.0-beta1
+message FilterSubscribeRequest {
+  enum FilterSubscribeType {
+    SUBSCRIBER_PING = 0;
+    SUBSCRIBE = 1;
+    UNSUBSCRIBE = 2;
+    UNSUBSCRIBE_ALL = 3;
   }
+
+  string request_id = 1;
+  FilterSubscribeType filter_subscribe_type = 2;
+
+  // Filter criteria
+  optional string pubsub_topic = 10;
+  repeated string content_topics = 11;
 }
 
-message MessagePush {
-  repeated WakuMessage messages = 1;
+message FilterSubscribeResponse {
+  string request_id = 1;
+  uint32 status_code = 10;
+  optional string status_desc = 11;
 }
 
-message FilterRPC {
-  string requestId = 1;
-  FilterRequest request = 2;
-  MessagePush push = 3;
+// Protocol identifier: /vac/waku/filter-push/2.0.0-beta1
+message MessagePush {
+  WakuMessage waku_message = 1;
 }
```

Comment:
What do you think of changing `MessagePush` to
```proto
message MessagePush {
  map<string, WakuMessages> messages_per_pubsubtopic = 20;
}

message WakuMessages {
  repeated waku.message.v1.WakuMessage messages = 1;
}
```

Similar to the scenario described [here](https://github.com/vacp2p/waku/pull/10#discussion_r1044389023), a problem we have right now is that it's not possible with this protobuffer to determine in which PubsubTopic was a `WakuMessage` received in case the full node is subscribed to more than one topic, so a subscriber that sent a `FilterSubscribeRequest` with an empty `pubsub_topic` could potentially receive duplicated `WakuMessage`

Reply:
Hah. Indeed. Thanks!

Note that there are ongoing discussions re exposing content topics vs pubsub topics (e.g [here](https://github.com/waku-org/waku.guide/pull/24#discussion_r1044368641)), but that does not obviate the need for associating WakuMessages with pubsub topics for Filter.

Reply:
I've added the `pubsub_topic` in https://github.com/vacp2p/rfc/pull/562/commits/d2d80b67ce26aef2b66877c3d5ad4813b7f755ba

LMK if this resolves your concern. :)

Reply:
Since I learned that we're sending a `MessagePush` per message, my suggestion is updated to this one instead :)
```proto
message MessagePush {
  WakuMessage waku_message = 1;
  string pubsub_topic = 2;
}
```

Reply:
Ah :) you're too fast and had already pushed the change! haha

Reply:
It does, thank you!